book/src/gobject_subclassing.md

```diff
@@ -3,44 +3,50 @@
 GObjects rely heavily on inheritance.
 Therefore, it makes sense that if we want to create a custom GObject, this is done via subclassing.
 Let's see how this works by replacing the button in our "Hello World!" app with a custom one.
-
 First, we need to create an implementation struct that holds the state and overrides the virtual methods.
-It is advised to keep it in a private module, since its state and methods are only meant to be used by the GObject itself.
-It therefore corresponds to the private section of objects in languages like Java and C++.
 
 <span class="filename">Filename: listings/gobject_subclassing/1/custom_button/imp.rs</span>
 
 ```rust ,no_run,noplayground
 {{#rustdoc_include ../listings/gobject_subclassing/1/custom_button/imp.rs}}
 ```
 The description of the subclassing is in `ObjectSubclass`.
-- `NAME` should consist of crate-name, module-path and object-name in order to avoid name collisions. Use [PascalCase](https://wiki.c2.com/?PascalCase) here.
+- `NAME` should consist of crate-name and object-name in order to avoid name collisions. Use [UpperCamelCase](https://en.wikipedia.org/wiki/Camel_case) here.
 - `Type` refers to the actual GObject that will be created afterwards.
 - `ParentType` is the GObject we inherit of.
 
 After that, we would have the option to override the virtual methods of our ancestors.
 Since we only want to have a plain button for now, we override nothing.
 We still have to add the empty `impl` though.
-Next, we describe our custom GObject.
+Next, we describe the public interface our custom GObject.
 
 <span class="filename">Filename: listings/gobject_subclassing/1/custom_button/mod.rs</span>
 
 ```rust ,no_run,noplayground
 {{#rustdoc_include ../listings/gobject_subclassing/1/custom_button/mod.rs:mod}}
 ```
 
-[`glib::wrapper!`](http://gtk-rs.org/gtk-rs-core/stable/latest/docs/glib/macro.wrapper.html) does the most of the work of subclassing for us.
-Coming from most other languages you would probably expect that you only have to mention the base class you want to inherit from.
-However, as of today, subclassing of GObjects in Rust requires to mention all ancestors and interfaces apart from `GObject` and `GInitiallyUnowned`.
-For `gtk::Button`, we can look them up in the corresponding [doc page](https://docs.gtk.org/gtk4/class.Button.html#hierarchy) of GTK4.
+[`glib::wrapper!`](http://gtk-rs.org/gtk-rs-core/stable/latest/docs/glib/macro.wrapper.html) implements the same traits that our `ParentType` implements.
+Theoretically that would mean that we `ParentType` is also the only thing we have to specify here.
+Unfortunately, nobody found yet a good way to do that.
```

Comment:
maybe it is worth mentioning that the list of types you provide for `@extends` / `@implements` is used to implement various traits for you like the IsA. That is why not mentioning everything is fine, until you need some specific methods from the `IsA<T>` implementation or you have to pass the new type to a method that takes `IsA<T>`.

Reply:
I did mention that you don't have to specify everything, so I assume you want me to be more precise here.
Correct?